update readme with pip information and basic example of prediction

pdan93 opened the pull request:
## Description
Update Readme with pip information.
Add basic example of downloading data and model and running a prediction.

Fixes #107 

## Checklist

- [x] Linked to Notion or GitHub issue
- [ ] Added tests
- [x] Updated README
- [ ] DVC up to date

nsorros commented on README.md:
````diff
@@ -32,11 +32,17 @@ at the moment.
+pip install grants-tagger
+```
+
+Alternatively you can clone this repository and build the project with 
````
This would not install the project, this would create the development environment for which we give instructions later. The manual installation would be to clone the project and run `pip install setup.py`

nsorros commented on README.md:
```diff
@@ -599,3 +604,23 @@ deploy               Deploy wheel to public s3 bucket
+
+Data can be downloaded from here after you make an account: http://participants-area.bioasq.org/datasets/
+Download from the `Datasets for task a` section, the latest available txt format dataset. This will result in a zip file what you can unzip, preferably in the `data\raw` folder. 
+You can download pretrained model from the releases section of this repository. For the current example we'll a model trained with the `mesh-cnn` approach: https://github.com/wellcometrust/grants_tagger/releases/download/v0.1.3/disease_mesh_cnn-2021.03.1.tar.gz
```
Note that the latest model could not be added to a release cause it was bigger than 2GB. So the standard way to download a model would be to use the CLI `grants_tagger download mesh`. That said assuming someone just wants to download the model we should add link that is updated in the README that either links to the release URL or s3, the link is the one in the code to download the model, this is just a manual way to get it.

pdan93 replied:
problem here is that it seems you need credentials to download from s3? I tried here to offer the public version.

nsorros replied:
you should not need credentials so if you do, I need to fix that. its shared in a public bucket

pdan93 replied:
yeah, seems this worked:

pdan93 replied:
`grants_tagger download epmc-mesh models/`

pdan93 replied:
`grants_tagger download model mesh`
seems to have worked, so no need. I'll update the README

nsorros commented:
⚠️ Note that I switched to main for the base branch. I switched the base branch on the PR but for tests to work you need to cherry pick 469b116038bedae879f1ab7068832d198cf12f1d. Just run `git cherry-pick 469b116038bedae879f1ab7068832d198cf12f1d` if you aim to do more commits and want to run tests against latest main and not the soonoutdated master

nsorros commented:
Is this indeed ready for review @pdan93?

pdan93 commented:
> Is this indeed ready for review @pdan93?

yes @nsorros

nsorros commented on README.md:
```diff
@@ -622,3 +627,28 @@ deploy               Deploy wheel to public s3 bucket
+
+We'll go through an example on how to download data and models, and use them for prediction and training using the cli commands.
+
+Data can be downloaded from here after you make an account: http://participants-area.bioasq.org/datasets/
```
I think data are not needed for the basic user. My idea is that the basic user is not trying to use the data to train or anything but apply to their own data. They want to know how to download a model and use it to make predictions.

nsorros commented:
Actually to start with I like that the section on using the production model is small. I suggest we move it at the top and include it in the install and use section, not create a new section

nsorros commented:
Also dont forget to update the section Bring your own model

nsorros commented:
As I was re reading the README I noticed two additional discrepancies that we should fix
* in predict, it mentions a tag command that does not exist anymore, we should remove the mention
* in bting your own_data it mentions `__main__` which is not `cli`, we should update
@pdan93

pdan93 commented:
I moved things to the install and use section, and cleaned it, and updated bring your own data. 
But the 'bring your own model' section, after looking at the code seems to be correct, what am I missing @nsorros ?

nsorros commented:
> I moved things to the install and use section, and cleaned it, and updated bring your own data.
> But the 'bring your own model' section, after looking at the code seems to be correct, what am I missing @nsorros ?

You are right, it looks good. Let me do a final review

nsorros reviewed (approved):
I am happy to merge as is but ideally there would be a Quickstart section that would demonstrate how someone can use the package or library in few commands to get predictions

aCampello commented:
Is this going to be merged?
